### PR TITLE
Moves special hug interactions that rely on bodyparts and special organs into signals

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Made this so @Thunder12345 could see how to add special item interactions with hugs without clogging up the help chain with more if else checks on niche items. And that's what I do now!

## Why It's Good For The Game

The if else chain in help act should only increase when base human things happen, we can now have special interactions for certain heads and tails... could be cool!

## Changelog

:cl:
refactor: Refactored hugging code, make an issue if you notice anything weird but it should be the same!
/:cl:

### CODE REVIEW

**Review by Kapu1178 — APPROVED** [review]

File: code/modules/surgery/organs/external/tails.dm [hunk 1]
```diff
@@ -56,7 +57,26 @@
...
+	to_chat(helper, span_notice("You pull on [helped]'s tail!"))
+	to_chat(helped, span_notice("[helper] pulls on your tail!"))
+	if(HAS_TRAIT(helped, TRAIT_BADTOUCH)) //How dare they!
+		to_chat(helper, span_warning("[helped] makes a grumbling noise as you pull on [p_their()] tail."))
+	else
+		helped.add_mood_event("tailpulled", /datum/mood_event/tailpulled)
+	return COMPONENT_SPECIAL_INTERACTION
+
 /obj/item/organ/external/tail/proc/wag(mob/user, start = TRUE, stop_after = 0)
+	SIGNAL_HANDLER
```
Comment (tralezab) [inline code/modules/surgery/organs/external/tails.dm:78]: imagine being a signal but NOT having signal handler could not be me

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/clothing/head/animalears.dm [hunk 2]
```diff
@@ -17,6 +28,23 @@
...
 
+///signal that fires when the owner is hugged, for special tail pulling (oh no) interactions...?
+/obj/item/clothing/head/costume/kitty/proc/on_owner_hug(mob/living/carbon/helped, mob/living/carbon/helper)
+	SIGNAL_HANDLER
+
+	if(helper.zone_selected != BODY_ZONE_PRECISE_GROIN)
+		return
+
+	helper.visible_message(span_danger("[helper] pulls on [helped]'s tail... and it rips off!"), \
+				null, span_hear("You hear a ripping sound."), DEFAULT_MESSAGE_RANGE, list(helper, helped))
```
Comment (Mothblocks) [inline code/modules/clothing/head/animalears.dm:39]:
```suggestion
		null, span_hear("You hear a ripping sound."), DEFAULT_MESSAGE_RANGE, list(helper, helped))
```

Only ever indent up one, no tab alignment

Comment (Mothblocks) [inline code/modules/clothing/head/animalears.dm:39]: But probably this:

```suggestion
	helper.visible_message(
		span_danger("[helper] pulls on [helped]'s tail... and it rips off!"), \
		null,
		span_hear("You hear a ripping sound."),
		DEFAULT_MESSAGE_RANGE,
		list(helper, helped),
	)
```

File: code/modules/surgery/bodyparts/head.dm [hunk 3]
```diff
@@ -311,6 +317,22 @@
...
 
+///signal that fires when the owner is hugged, for special head pat interactions!
+/obj/item/bodypart/head/proc/on_owner_hug(mob/living/carbon/helped, mob/living/carbon/helper)
+	SIGNAL_HANDLER
+
+	if(check_zone(helper.zone_selected) != BODY_ZONE_HEAD)
+		return
+
+	helper.visible_message(span_notice("[helper] gives [helped] a pat on the head to make [p_them()] feel better!"), \
+				null, span_hear("You hear a soft patter."), DEFAULT_MESSAGE_RANGE, list(helper, helped))
```
Comment (Mothblocks) [inline code/modules/surgery/bodyparts/head.dm:328]: Same

File: code/modules/surgery/organs/external/tails.dm [hunk 4]
```diff
@@ -56,7 +57,26 @@
...
 
+///signal that fires when the owner is hugged, for special tail pulling interactions!
+/obj/item/organ/external/tail/proc/on_owner_hug(mob/living/carbon/helped, mob/living/carbon/helper)
+	SIGNAL_HANDLER
+
+	if(helper.zone_selected != BODY_ZONE_PRECISE_GROIN)
+		return
+
+	helper.visible_message(span_notice("[helper] pulls on [helped]'s tail!"), \
+				null, span_hear("You hear a soft patter."), DEFAULT_MESSAGE_RANGE, list(helper, helped))
```
Comment (Mothblocks) [inline code/modules/surgery/organs/external/tails.dm:68]: Same

**Review by Time-Green — APPROVED** [review]

**Review by Kapu1178 — APPROVED** [review]

**Review by Time-Green — APPROVED** [review]

**Review by Kapu1178 — APPROVED** [review]

**Review by Time-Green — APPROVED** [review]

**Review by Kapu1178 — APPROVED** [review]

**Review by Time-Green — APPROVED** [review]

**Review by Kapu1178 — APPROVED** [review]

**Comment by Mothblocks** [comment]
stop

**Review by Fikou — APPROVED** [review]

**Comment by tralezab** [comment]
ban that guy for at least 5 minutes

**Comment by Mothblocks** [comment]
As we have an abnormally large number of open PRs, I am closing all ones that have been marked stale. Please reopen the PR when it is ready and you have an interested maintainer

**Pull request closed without merging**